src/main/java/com/conveyal/taui/models/ModificationStop.java

```diff
@@ -90,63 +85,100 @@ private ModificationStop (Coordinate c, String id, double distanceFromStart, boo
 
         if (firstSegment.stopAtStart) {
             stops.add(new ModificationStop(firstSegment.geometry.getCoordinates()[0],
-                    firstSegment.fromStopId, 0, false));
+                    firstSegment.fromStopId, false, 0));
         }
 
-        double distanceToLastStop = 0; // distance to previously created stop, from start of pattern
-        double distanceToLineSegmentStart = 0; // from start of pattern
-        for (Segment segment : segments) {
+        double metersToLastStop = 0; // distance to previously created stop, from start of pattern
+        double metersFromPatternStart = 0; // from start of pattern
+        int hopTimeSeconds = 0;
+        for (int i = 0; i < segments.size(); i++) {
+            Segment segment = segments.get(i);
+            int segmentSpeedKph = i < segmentSpeedsKph.length ? segmentSpeedsKph[i] : DEFAULT_SEGMENT_SPEED_KPH;
```

Comment:
Since segments are owned by the modification and speeds are owned by the time table we've "allowed" the lengths to get out of sync. Setting them to a default speed may not be the best behavior. Intuitively I'd think we should use the previous segment speed by default. So if there are 5 segments and 3 segment speeds, the last two segments would have the same speed as the third segment. Does that make sense?

Reply:
Does https://github.com/conveyal/analysis-ui/pull/854 mean the UI should always save arrays of equal length for `segments` and `segmentSpeedsKph`?

Reply:
No. Even with the latest changes it will not be guaranteed unless we do a database migration.